internal/server/server.go

```diff
@@ -335,7 +337,10 @@ func (s *Server) DelUserURLs(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	ok, err := s.service.DelURLs(cookie.Value, buf)
+	go func() {
+		_, err = s.service.DelURLs(cookie.Value, buf)
```

Comment:
```suggestion
		_, err := s.service.DelURLs(cookie.Value, buf)
```
Не следует делать в этой функции зацепление на err из хендлера. Если хочется что-то сделать с ошибкой, необходимо ее в этой функции создать и, например, залогировать. Функция все равно закончит свою работу уже после того, как хендлер закроется.

Зато в данную функцию необходимо как-то доставить WaitGroup из main (например, поместить его в объект Server при инициализации). Потому что при шатдауне необходимо сначала дождаться завершения работы всех этих горутин и только потом закрывать БД. В текущей реализации велика вероятность попасть на панику и потерю данных.

Получается, что fan-in реализован теперь верно, но graceful shutdown все еще не доделан.

Reply:
Убрал WaitGroup. Синхронизация теперь только на атомик-счетчиках, которые перенес из SetDelBatch в саму DelURLs. Теперь когда в хэндлере DelUserURLs запускается горутина с DelURLs то в самом начале остановка приложения будет тормозиться инкрементом атомик-счетчика. После завершения всех функций (SetDelBatch в случае БД) декрементируется атомик-счетчик и отсылается уведомление через канал в Run(). Тогда если атомик-счетчик равен нулю, закрывается БД и завершается приложение.